### PR TITLE
[Docs] Add doc on setjmp-longjmp support

### PR DESCRIPTION
**Pull request by aheejin**
This adds a doc on setjmp-longjmp support. This also describes how to use exceptions and setjmp-longjmp together and restrictions that apply.

While we have been allowing Wasm EH + Emscripten SjLj combination as an interim solution while Wasm SjLj support was not implemented yet, this only documents Emscripten EH + Emscripten SjLj and Wasm EH + Wasm SjLj combinations, to reduce confusion. (Emscripten EH is what we call "JavaScript-based support" in the docs) Wasm EH + Emscripten SjLj has a lot more restrictions, including you cannot even use `setjmp` in the same function that uses `try`-`catch` or stack allocated objects that have destructors, and we don't intend to maintain support for that combination anyway.

### CODE REVIEW

File: site/source/docs/porting/setjmp-longjmp.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
+.. _setjmp-longjmp:
+
+========================
+C setjmp-longjmp Support
+========================
+
+setjmp-longjmp support is enabled by default in Emscripten. This is controlled
+by ``SUPPORT_LONGJMP`` setting, which can take these values:
```
Comment (kripken) [inline site/source/docs/porting/setjmp-longjmp.rst:8]:
```suggestion
by the ``SUPPORT_LONGJMP`` setting, which can take these values:
```

File: site/source/docs/porting/setjmp-longjmp.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+- ``emscripten``: JavaScript-based support
+- ``wasm``: WebAssembly exception handling-based support
+- 0: No support
+- 1: Default support (currently ``emscripten``) (default)
+
+Currently ``-sSUPPORT_LONGJMP=1`` is the same as
+``-sSUPPORT_LONGJMP=emscripten``, and turned on by default. This will later be
```
Comment (kripken) [inline site/source/docs/porting/setjmp-longjmp.rst:16]:
```suggestion
``-sSUPPORT_LONGJMP=emscripten``, and turned on by default. This default will eventually be
```

File: site/source/docs/porting/setjmp-longjmp.rst [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+``-sSUPPORT_LONGJMP=1``, which defaults to ``emscripten``, is enabled by
+default, so you don't need to pass it explicitly.
+
+To use the WebAssembly EH and setjmp-longjmp support together:
+
+.. code-block:: bash
+
+  em++ -fwasm-exceptions -sSUPPORT_LONGJMP=wasm test.cpp -o test.js
+
+There is one restriction for using WebAssembly EH-based support for exceptions
```
Comment (kripken) [inline site/source/docs/porting/setjmp-longjmp.rst:90]:
```suggestion
There is one specific restriction for using WebAssembly EH-based support for exceptions
```

File: site/source/docs/porting/setjmp-longjmp.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+and setjmp-longjmp at the same time. You cannot call ``setjmp`` within a C++
+``catch`` clause. For example, the following will not work:
```
Comment (kripken) [inline site/source/docs/porting/setjmp-longjmp.rst:92]: Does this error at compile time or runtime? Might be worth mentioning.

**Review by kripken — APPROVED** [review]
lgtm % comments

File: site/source/docs/porting/setjmp-longjmp.rst (hunk 4 above)
Comment (aheejin) [reply site/source/docs/porting/setjmp-longjmp.rst:92]: Compile time. Will specify.

**Review by kripken — APPROVED** [review]

**Pull request merged**